pkg/kube/client.go

```diff
@@ -369,13 +370,55 @@ func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<-
 			kind = currentKind
 		}
 		wg.Add(1)
-		go func(i *resource.Info) {
-			errs <- fn(i)
+		go func(info *resource.Info) {
+			// try 3 times total
+			tries := 3
```

Comment:
can it be changed from the outside?

Reply:
you mean move t`ries := 3` 2 lines up to right after wg.Add(1)?

Reply:
I am just thinking about, for example, it is changed to `func batchPerform(infos ResourceList, fn func(context.Context, *resource.Info) error, errs chan<- error)`

Reply:
I hope I understood you right. Now, it looks like this:
~~~
func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<- error) {
        var kind string
        var wg sync.WaitGroup
        for _, info := range infos {
                currentKind := info.Object.GetObjectKind().GroupVersionKind().Kind
                if kind != currentKind {
                        wg.Wait()
                        kind = currentKind
                }
                wg.Add(1)

                quotaConflictRetries := 3
                go func(info *resource.Info) {
~~~